### PR TITLE
Fix element-wise lowering

### PR DESCRIPTION
**Pull request by makslevental**
Disabling vectorization at the LLVM level did the trick.

Fixes https://github.com/nod-ai/iree-amd-aie/issues/594

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEVectorization.cpp [hunk 1]
```diff
@@ -77,6 +77,7 @@ void AMDAIEVectorizationPass::runOnOperation() {
     // dominate this use'. TODO(newling) follow-up on this.
     if (isa<linalg::FillOp>(op)) return;
 
+    if (isElementwise(cast<linalg::LinalgOp>(op))) return;
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEVectorization.cpp:80]: I know this is originally written by me, but could you add some comments above to explain why we disable the elementwise op for vectorization?

File: build_tools/ci/cpu_comparison/test_files/matmul_elem_2.mlir [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        %15 = arith.addf %13, %cst_1 : f32
+        %16 = arith.cmpf ult, %15, %cst_2 : f32
+        %17 = arith.cmpf ugt, %15, %cst_3 : f32
+        %18 = arith.select %16, %cst_2, %15 : f32
+        %19 = arith.select %17, %cst_3, %18 : f32
+        %20 = arith.fptosi %19 : f32 to i8
+        linalg.yield %20 : i8
+    } -> tensor<4240x160xi8>
+  return %9 : tensor<4240x160xi8>
+}
```
Comment (yzhang93) [inline build_tools/ci/cpu_comparison/test_files/matmul_elem_2.mlir:32]: Add an empty line.

**Comment by yzhang93** [comment]
I don't understand which codes did you change to disable vectorization at the LLVM level?

**Review by newling — CHANGES_REQUESTED** [review]
The changes to the testing are completely unrelated to this PR and should be done in a separate PR

**Comment by newling** [comment]
> The changes to the testing are completely unrelated to this PR and should be done in a separate PR

Sorry, they're not all unrelated, I didn't realise you were touching the test file anyway by adding a new test

**Review by newling — APPROVED** [review]

**Comment by makslevental** [comment]
> I don't understand which codes did you change to disable vectorization at the LLVM level?

That happened here https://github.com/nod-ai/iree-amd-aie/pull/622

**Comment by newling** [comment]
Might make sense to add a test here: https://github.com/nod-ai/iree-amd-aie/blob/cd9138696b5bdb0e804af81a7c1b1013664113e7/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/vectorization.mlir#L48

**Pull request merged**